moon/controller.py

```diff
@@ -118,14 +124,38 @@ def send_speed_cmd(self, speed, angular_speed):
             self.virtual_bumper.update_desired_speed(speed, angular_speed)
         self.bus.publish('desired_speed', [round(speed*1000), round(math.degrees(angular_speed)*100)])
 
-    def send_request(self, cmd):
-        """Send ROS Service Request form a single place"""
-        self.publish('request', cmd)
-        while True:
-            dt, channel, data = self.listen()
-            if channel == 'response':
-                return data
-            print(dt, 'ignoring', channel)
+    def on_response(self, timestamp, data):
+        intoken, response = data
+        self.requests[intoken] = {
+            "response": response
+            }
+
+    def poll_responses(self, token, callback):
+        while self.time is None:
+            time.sleep(0.1)
```

Comment:
use `self.bus.sleep(0.1)` for general module or `self.sleep(0.1)` for class inherited from `Node`